examples/spmm/spmm.cc

```diff
@@ -627,7 +638,7 @@ static int parseOption(std::string &option, int default_value) {
     pos = option.length();
   }
   token = option.substr(0, pos);
-  N = std::stoi(token);
+  N = std::stol(token);
```

Comment:
I think if parseOption is used to parse an unsigned int this will have the same truncating effect as using strtoi: N, as a signed int cannot store the seed.

My fix consisted in providing a static long parseOption(... long default_value), and upgrade the seed to a long. This should be in this branch already.